src/routes/explore/+page.server.js

```diff
@@ -0,0 +1,44 @@
+/** @type {import('./$types').PageServerLoad} */
+export async function load({ fetch, url }) {
+	const itemsPerPage = url.searchParams.get("limit") || 6;
+	const currentPage = url.searchParams.get("page") || 1;
+
+	async function fetchAllProjects() {
+		try {
+			const response = await fetch(
+				`/api/projects/allprojects?page=${currentPage}&limit=${itemsPerPage}`,
+			);
+			if (!response.ok) {
+				return [];
+			}
+
+			return await response.json();
+		} catch (error) {
+			// TODO: do something with the error
+		}
+	}
+
+	async function fetchTopProjects() {
+		try {
+			const response = await fetch("/api/projects/allprojects?limit=3");
+			if (!response.ok) {
+				return [];
+			}
+
+			return await response.json();
+		} catch (error) {
+			// TODO: do something with the error
+		}
+	}
```

Comment:
could you move this outside the load function.

for errors you can just return the accurate http status code